gelic-idealab/impress
# Upgrade Komodo Package to v0.5.4

# Upgrade Komodo Package to v0.5.4

## Cleaned up and refactored code, adjusted colors

Commented out code related to Primitives and Grouping features, which will come later. Made ImpressMenu2 panels into prefabs. Adjusted colors.

### Type of Change


### How Has This Been Tested?

Visually inspect menu for coloring consistency.

## Repaired eraser

Fixed bug where eraser would only be enabled for the left hand.

See more details in #1 

### Type of change


### How Has This Been Tested?

Steps to test: 

1. Enter Play mode. 
2. Make sure the Scene and Game windows are visible at the same time
3. Select ManualTestHelper in the hierarchy.
4. Right click the script component on it and choose "Switch to XR"
5. Move handL so that it points at the Create tab
6. In the Game window, press the T key to open the Create panel
7. Use the same technique to press the Erase button

Expected result: both eraser displays show up.


## Added ManualTestHelper and ControllersManager

Created ManualTestHelper script and gameObject. To use it, right click the script component on the gameObject in the Main scene and select a test function.

I also introduced ControllersManager, which is a future candidate for refactoring KomodoControllerInteraction.cs. For now, it supports the `T` and `Y` keys, which correspond to the left controller trigger and the right controller trigger. To use this, make sure the game window is focused and the menu mode is set to XR mode.

### Type of change


### How Has This Been Tested?

See the test for "Repaired Eraser" above.

Expected result: holding `T` activates left selection, drawing, or erasing, and holding `Y` activates right selection, drawing, or erasing


## Added ManualTestHelper and ControllersManager

Created ManualTestHelper script and gameObject. To use it, right click the script component on the gameObject in the Main scene and select a test function.

I also introduced ControllersManager, which is a future candidate for refactoring KomodoControllerInteraction.cs. For now, it supports the `T` and `Y` keys, which correspond to the left controller trigger and the right controller trigger. To use this, make sure the game window is focused and the menu mode is set to XR mode.

## Repaired ColorPicker color selection

Color selection now reflects the correct color rather than black or white. It also works on the first opening of the draw tab, unlike before. 

Under the hood, ColorPicker has been reworked into a prefab. Most functions have been moved to a static class, ColorPickerManager. The script component on the ColorImage has been reduced to the essentials. Both classes are initialized regardless of whether the draw tab is open thanks to a new script, CreateMenu.

See more details in #1 


### How Has This Been Tested?

See the test for "Repaired Eraser" above for how to use the VR hands in the editor on a screen.

Expected result: selecting the draw tab results in no errors, the selected color is initially white, hovering over the ColorImage results in a correct preview color, and selecting a color results in the correctly displayed color, and drawing with that color results in the correct draw stroke color.

____

## Test Configuration

No browser yet
No VR headset yet

# Checklist:


Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).
